src/style/style.js

```diff
@@ -1795,9 +1837,17 @@ class Style extends Evented {
         return this._numCircleLayers > 0;
     }
 
-    clearWorkerCaches() {
+    _clearWorkerCaches() {
         this.dispatcher.broadcast('clearCaches');
     }
+
+    destroy() {
+        this._clearWorkerCaches();
+        if (this.terrainSetForDrapingOnly()) {
```

Comment:
The way we are creating and deleting "terrain" for draping seems a bit brittle. Do you see any way of cleaning this up a bit without a huge refactor?

Reply:
Not at the moment, the proper way to handle this would be to port the 2d render cache support from native and have a draping renderer interface that can be used on and off independently of whether you need terrain, which in itself would possibly be as large as a task as the globe.